src/scaladoc/scala/tools/nsc/doc/base/comment/Body.scala

```diff
@@ -86,7 +86,9 @@ final case class HtmlTag(data: String) extends Inline {
   }
 
   private val TagsNotToClose = Set("br", "img")
-  def close = tagName collect { case name if !TagsNotToClose(name) => HtmlTag(s"</$name>") }
+  def close = tagName collect {
+    case name if !TagsNotToClose(name) && !data.endsWith(s"</$name>") => HtmlTag(s"</$name>")
+  }
```

Comment:
The change looks correct - the usage of `HtmlTag` a bit messy in general... Since you are in the area, would you mind pulling out `Pattern` and `TagsNotToClose` into the companion of `HtmlTag`? There's no reason to have them as fields in every instance.